addons/mail/static/src/discuss/core/public_web/discuss_search.js

```diff
@@ -0,0 +1,243 @@
+            },
+            () => [this.state.searchValue]
+        );
+    }
+
+    get inputPlaceholder() {
+        if (this.state.selectedPartners.length) {
+            return _t("Press Enter to start the conversation");
+        } else if (!this.props.category) {
+            return _t("Search channels and DMs");
+        } else {
+            return this.props.category.addTitle;
+        }
+    }
+
+    async fetchSuggestions() {
+        const cleanedTerm = cleanTerm(this.state.searchValue);
+        if (!cleanedTerm) {
+            return;
+        }
+        const data = await this.sequential(async () => {
+            this.state.isFetching = true;
+            const data = await rpc("/discuss/search", {
+                term: cleanedTerm,
+                category_id: this.props.category?.id,
+            });
+            this.state.isFetching = false;
+            return data;
+        });
+        this.store.insert(data);
+    }
+
+    get suggestions() {
+        const cleanedTerm = cleanTerm(this.state.searchValue);
+        const suggestions = [];
+        if (!cleanedTerm) {
+            return suggestions;
+        }
+        if (!this.props.category || this.props.category === this.store.discuss.chats) {
+            suggestions.push(
+                ...Object.values(this.store.Persona.records)
+                    .filter(
+                        (persona) =>
+                            persona !== this.store.self &&
+                            persona.isInternalUser &&
+                            cleanTerm(persona.name).includes(cleanedTerm)
+                    )
+                    .map((persona) => {
+                        return {
+                            optionTemplate: "discuss.DiscussSearch.partner",
+                            classList: "o-mail-DiscussSearch-suggestion",
+                            partner: persona,
+                            group: 10,
+                        };
+                    })
+            );
+            if (this.state.selectedPartners.length) {
+                return suggestions;
+            }
+        }
+        if (!this.props.category || this.props.category === this.store.discuss.channels) {
+            suggestions.push(
+                ...Object.values(this.store.Thread.records)
+                    .filter(
+                        (thread) =>
+                            thread.channel_type === "channel" &&
+                            cleanTerm(thread.name).includes(cleanedTerm)
+                    )
+                    .map((thread) => {
+                        return {
+                            optionTemplate: "discuss.DiscussSearch.channel",
+                            classList: "o-mail-DiscussSearch-suggestion",
+                            channel: thread,
+                            group: 90,
+                        };
+                    })
+            );
+        }
+        return suggestions;
+    }
```

Comment:
Should also take anonymous visitors into account:
<img width="302" alt="Screenshot 2024-09-12 at 17 32 52" src="https://github.com/user-attachments/assets/ca7d9b2f-633c-4cf8-8185-030c0f8d1392">

_(whatsapp conversations seem to work, would be nice at least livechat and whatsapp test coverage)_